internal/drivers/nomad/driver.go

```diff
@@ -335,82 +351,86 @@ func (p *config) resourceJob(cpus, memGB int, vm, accountID string) (job *api.Jo
 	// of cleanup - as the job has created a virtual machine but we could not parse the node identifier.
 	l, _, err := p.client.Jobs().Allocations(id, false, nil)
 	if err != nil {
-		return ip, nodeID, port, err
+		return ip, nodeID, ports, err
 	}
 	if len(l) == 0 {
-		return ip, nodeID, port, errors.New("scheduler: no allocation found for the job")
+		return ip, nodeID, ports, errors.New("scheduler: no allocation found for the job")
 	}
 
 	nodeID = l[0].NodeID
 	allocID := l[0].ID
 	if nodeID == "" || allocID == "" {
-		return ip, nodeID, port, errors.New("scheduler: could not find an allocation identifier for the job")
+		return ip, nodeID, ports, errors.New("scheduler: could not find an allocation identifier for the job")
 	}
 
 	alloc, _, err := p.client.Allocations().Info(allocID, &api.QueryOptions{})
 	if err != nil {
-		return ip, nodeID, port, err
+		return ip, nodeID, ports, err
 	}
 
 	// Not expected - if nomad is unable to find a port, it should not run the job at all.
 	if alloc.Resources.Networks == nil || len(alloc.Resources.Networks) == 0 {
 		err = fmt.Errorf("scheduler: could not allocate network and ports for job")
 		logr.Errorln(err)
-		return ip, nodeID, port, err
+		return ip, nodeID, ports, err
 	}
 
-	port = alloc.Resources.Networks[0].DynamicPorts[0].Value
+	ports = map[string]int{}
+	for i := range alloc.Resources.Networks[0].DynamicPorts {
+		ports[alloc.Resources.Networks[0].DynamicPorts[i].Label] = alloc.Resources.Networks[0].DynamicPorts[i].Value
+	}
 
 	// sanity check
-	if port <= 0 || port > 65535 {
-		err = fmt.Errorf("scheduler: port %d generated is not a valid port", port)
-		logr.Errorln(err)
-		return ip, nodeID, port, err
+	for _, port := range ports {
+		if port <= 0 || port > 65535 {
+			err = fmt.Errorf("scheduler: port %d generated is not a valid ports", port)
+			logr.Errorln(err)
+			return ip, nodeID, ports, err
+		}
 	}
 
 	n, _, err := p.client.Nodes().Info(nodeID, &api.QueryOptions{})
 	if err != nil {
 		logr.WithError(err).Errorln("scheduler: could not get information about the node which picked up the resource job")
-		return ip, nodeID, port, err
+		return ip, nodeID, ports, err
 	}
 
 	ip = strings.Split(n.HTTPAddr, ":")[0]
 	if net.ParseIP(ip) == nil {
 		err = fmt.Errorf("scheduler: could not parse client machine IP: %s", ip)
 		logr.Errorln(err)
-		return ip, nodeID, port, err
+		return ip, nodeID, ports, err
 	}
 
-	return ip, nodeID, port, nil
+	return ip, nodeID, ports, nil
 }
 
 // initJob creates a job which is targeted to a specific node. The job does the following:
 //  1. Starts a VM with the provided config
 //  2. Runs a startup script inside the VM
-func (p *config) initJob(vm, startupScript string, hostPort int, nodeID string, resource cf.NomadResource) (job *api.Job, id, group string) {
+func (p *config) initJob(vm, startupScript string, liteEnginePort int, gitspacesPortMappings map[int]int, nodeID string, resource cf.NomadResource) (job *api.Job, id, group string) {
```

Comment:
Please test CI once to make sure there's no regression on the existing flow

Reply:
Ran automation locally. There are a few failures (github connector issue), but these are failing even on the main branch (without my changes). cc: @harshjain12